src/main/java/ax/commands/Parser.java

```diff
@@ -0,0 +1,120 @@
+package ax.commands;
+
+import ax.display.Ui;
+import ax.task.*;
+
+import java.io.FileWriter;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.MissingFormatArgumentException;
+import java.util.Scanner;
+
+public class Parser {
+    /**
+     * Gets input from the user and performs the corresponding action.
+     * If the input is "list", displays the current to-do list.
+     * If the input is "todo", adds a todo with specified text.
+     * If the input is "deadline", adds a deadline with specified text and /by for date.
+     * If the input is "event", adds an event with specified text and /from date and /to date.
+     * If the input is "delete", prompts the user for the index of the item to delete from the list.
+     * If the input is "mark", sets the item at specified index to true
+     * If the input is "unmark", sets the item at specified index to false
+     * If the input is anything else, throws a NoSuchMethodException.
+     * If the input is missing an argument, throws a MissingFormatArgumentException.
+     *
+     * @param scanner the scanner to use for input
+     * @return true if the user wants to exit, false otherwise
+     */
+    @SuppressWarnings("unused")
+    public static boolean getInput(Scanner scanner) {
```

Comment:
This method is quite long. Can consider splitting it with more smaller methods to handle each case.